packages/eas-cli/src/update/utils.ts

```diff
@@ -269,3 +274,90 @@ export function getBranchDescription(branch: UpdateBranchFragment): FormattedBra
     },
   };
 }
+
+export function isBundleDiffingEnabled(exp: ExpoConfig): boolean {
+  return (exp.updates as any)?.enableBsdiffPatchSupport === true;
+}
+
+// Make authenticated requests to the launch asset URL with diffing headers
+export async function prewarmDiffingAsync(
+  graphqlClient: ExpoGraphqlClient,
+  appId: string,
+  newUpdates: UpdatePublishMutation['updateBranch']['publishUpdateGroups']
+): Promise<void> {
+  const DUMMY_EMBEDDED_UPDATE_ID = '00000000-0000-0000-0000-000000000000';
+
+  const toPrewarm = [] as {
+    update: UpdatePublishMutation['updateBranch']['publishUpdateGroups'][0];
+    launchAssetKey: string;
+  }[];
+
+  for (const update of newUpdates) {
+    try {
+      const manifest = JSON.parse(update.manifestFragment);
+      const launchAssetKey: string | undefined = manifest.launchAsset?.storageKey;
+      const requestedUpdateId: string = update.id;
+      if (!launchAssetKey || !requestedUpdateId) {
+        continue;
+      }
+      toPrewarm.push({
+        update,
+        launchAssetKey,
+      });
+    } catch {
+      // ignore updates without launch assets
+    }
```

Comment:
what portion of the code in the `try` can fail? from my read it gracefully handles updates without launch assets so that wouldn't throw. I think the only thing that could throw is the JSON.parse?

I think prefer to not wrap in try/catch here.

```suggestion
    const manifest = JSON.parse(update.manifestFragment);
    const launchAssetKey: string | undefined = manifest.launchAsset?.storageKey;
    const requestedUpdateId: string = update.id;
    if (!launchAssetKey || !requestedUpdateId) {
      continue;
    }
    toPrewarm.push({
      update,
      launchAssetKey,
    });
```